app/models/eventdate.rb

```diff
@@ -172,25 +191,27 @@ def run_positions_for(member)
     self.event_roles.where(member: member)
   end
 
-  def self.runify(eventdates)
-    eventdates.chunk do |ed|
-      ed.full_roles
-    end.map do |roles, run|
-      run
-    end
+  def self.group_by_runcrew(eventdates)
+    # In the event list, we want to be able
+    # to share one runcrew list with multiple
+    # adjacent eventdate rows if they are identical
+    eventdates.chunk(&:full_roles).map { |roles, eds| eds }
```

Comment:
This does still work. I'm not particularly attached to either way of writing it, but I thought all the line breaking made it hard to read. Maybe it would be best to write the chunk with a one-line block? Up to you.